session/session.go

```diff
@@ -46,19 +65,26 @@ func decryptCookie(cookie []byte, secret []byte) (dd []byte, err error) {
 	return
 }
 
-func DecryptSignedCookie(signed_cookie, secret_key_base, salt string) (session []byte, err error) {
-	cookie, err := url.QueryUnescape(signed_cookie)
+func DecryptSignedCookie(signedCookie, secretKeyBase, salt, signSalt string) (session []byte, err error) {
+	cookie, err := url.QueryUnescape(signedCookie)
 	if err != nil {
 		return
 	}
 
 	vectors := strings.SplitN(cookie, "--", 2)
+	if vectors[0] == "" || vectors[1] == "" {
+		return nil, errors.New("Invalid cookie!")
```

Comment:
Let's use lower case and prepend the error string with package name to follow the [proposed error format](https://blog.golang.org/error-handling-and-go). Also I'm not sure if we should use `!` here.